cmd/kube-apiserver/app/server.go

```diff
@@ -447,38 +416,22 @@ func Run(s *options.APIServer) error {
 		}
 	}
 
+	genericConfig := genericapiserver.NewConfig(s.ServerRunOptions)
+	// TODO: Move the following to generic api server as well.
+	genericConfig.StorageDestinations = storageDestinations
+	genericConfig.StorageVersions = storageVersions
+	genericConfig.Authenticator = authenticator
+	genericConfig.SupportsBasicAuth = len(s.BasicAuthFile) > 0
+	genericConfig.Authorizer = authorizer
+	genericConfig.AdmissionControl = admissionController
+	genericConfig.APIResourceConfigSource = apiResourceConfigSource
+	genericConfig.MasterServiceNamespace = s.MasterServiceNamespace
+	genericConfig.ProxyDialer = proxyDialerFn
+	genericConfig.ProxyTLSClientConfig = proxyTLSClientConfig
+	genericConfig.Serializer = api.Codecs
```

Comment:
That's not appropriate - not all generic api servers will have api.Codecs.  Generic API server should not require a particular runtime.Scheme